iree/samples/static_library/CMakeLists.txt

```diff
@@ -92,3 +94,85 @@ iree_lit_test(
   LABELS
     "hostonly"
 )
+
+## Example with VM C module.
+# Setup args for iree-translate.
+set(_TRANSLATE_ARGS)
+list(APPEND _TRANSLATE_ARGS "-iree-mlir-to-vm-c-module")
+list(APPEND _TRANSLATE_ARGS "-iree-hal-target-backends=dylib-llvm-aot")
+list(APPEND _TRANSLATE_ARGS "-iree-llvm-link-embedded=false")
+list(APPEND _TRANSLATE_ARGS "-iree-llvm-link-static")
+list(APPEND _TRANSLATE_ARGS "-iree-llvm-static-library-output-path=simple_mul_c_module.o")
+list(APPEND _TRANSLATE_ARGS "${CMAKE_CURRENT_SOURCE_DIR}/simple_mul.mlir")
+list(APPEND _TRANSLATE_ARGS "-o")
+list(APPEND _TRANSLATE_ARGS "simple_mul_emitc.h")
+
+# Custom command for iree-translate to generate static library and C module.
+add_custom_command(
+  OUTPUT
+    ${CMAKE_CURRENT_BINARY_DIR}/simple_mul_c_module.h
+    ${CMAKE_CURRENT_BINARY_DIR}/simple_mul_c_module.o
+    ${CMAKE_CURRENT_BINARY_DIR}/simple_mul_emitc.h
+  COMMAND ${_TRANSLATE_TOOL_EXECUTABLE} ${_TRANSLATE_ARGS}
+  DEPENDS ${_TRANSLATE_TOOL_EXECUTABLE} "simple_mul.mlir"
+)
+
+# TODO(marbre): Cleanup custom targets and libraries.
+add_custom_target(
+  simple_mul_gen
+  DEPENDS
+    "simple_mul_emitc.h"
+)
+
+add_library(simple_mul_c_module
+ STATIC
+ ${CMAKE_CURRENT_BINARY_DIR}/simple_mul_c_module.h
+ ${CMAKE_CURRENT_BINARY_DIR}/simple_mul_c_module.o
+)
+
+SET_TARGET_PROPERTIES(
+  simple_mul_c_module
+  PROPERTIES
+  LINKER_LANGUAGE C
+)
+
+add_library(simple_mul_h STATIC "")
+target_sources(simple_mul_h PRIVATE "simple_mul_c_module.h")
+add_dependencies(simple_mul_h simple_mul_gen)
+
+SET_TARGET_PROPERTIES(
+ simple_mul_h
+ PROPERTIES
+ LINKER_LANGUAGE C
+)
+
+# TODO(marbre): Check sources and dependencies.
+iree_cc_binary(
+NAME
+  static_library_demo_c
+SRCS
+  "create_c_module.c"
+  "static_library_demo.c"
+  "simple_mul_c_module.h"
+  "simple_mul_emitc.h"
+DEPS
+  iree::runtime
+  iree::hal::local::loaders::static_library_loader
+  iree::hal::local::sync_driver
+  iree::task::api
```

Comment:
Don't need to include the `iree::task::api` library since we don't use the task_executor after #7571. Can fix `static_library_demo` target with another commit.

Reply:
Excellent catch! We need to add an `if(IREE_ENABLE_EMITC)` anyway. Thus, we're going to remove that dep when touching the CMakeLists.